tests/DialogServiceConnectorTests.ts

```diff
@@ -1112,6 +1114,46 @@ test("SendActivity fails with invalid JSON object", (done: jest.DoneCallback) =>
     });
 });
 
+describe('Agent config message tests', () => {
+    let eventListener: IDetachable;
+    let observedAgentConfig: AgentConfig;
+
+    beforeEach(() => {
+        eventListener = Events.instance.attachListener({
+            onEvent: (event: PlatformEvent) => {
+                if (event instanceof SendingAgentContextMessageEvent) {
+                    const agentContextEvent = event as SendingAgentContextMessageEvent;
+                    observedAgentConfig = agentContextEvent.agentConfig;
+                }
+            },
+        });
+    });
+
+    afterEach(async () => {
+        await eventListener.detach();
+        observedAgentConfig = undefined;
+    });
+
+    test('Agent connection id can be set', async (done: jest.DoneCallback) => {
+        const testConnectionId: string = "thisIsTheTestConnectionId";
+        const dialogConfig: sdk.BotFrameworkConfig = BuildBotFrameworkConfig();
+        dialogConfig.setProperty(sdk.PropertyId.Conversation_Agent_Connection_Id, testConnectionId);
+        objsToClose.push(dialogConfig);
+        const connector: sdk.DialogServiceConnector = BuildConnectorFromWaveFile(dialogConfig);
+        objsToClose.push(connector);
+
+        connector.listenOnceAsync(
+            () => {
+                expect(observedAgentConfig).not.toBeUndefined();
+                expect(observedAgentConfig.get().botInfo.connectionId === testConnectionId);
+                done();
```

Comment:
Because this is in a callback, you want a try / catch / fail pattern here. Else an error will just show up as a timeout.